modules/cudaoptflow/src/pyrlk.cpp

```diff
@@ -339,8 +354,13 @@ namespace
         {
         }
 
-        virtual Size getWinSize() const { return winSize_; }
-        virtual void setWinSize(Size winSize) { winSize_ = winSize; }
+        virtual Size getWinSize() const { return cv::Size(winSize_[0], winSize_[1]); }
+        virtual void setWinSize(Size winSize) {
+            winSize_[0] = winSize.width;
+            winSize_[1] = winSize.height;
+            halfWinSize_[0] = (winSize.width - 1) / 2;
+            halfWinSize_[1] = (winSize.height -1) / 2;
```

Comment:
`halfWinSize_` is updated in three locations, but used in `loadWinSize` only.
Consider to drop this field completely and calculate it on-the-fly.